util/git/client.go

```diff
@@ -653,17 +658,28 @@ func (m *nativeGitClient) VerifyCommitSignature(revision string) (string, error)
 	return out, nil
 }
 
+// IsAnnotatedTag returns true if the revision points to an annotated tag
+func (m *nativeGitClient) IsAnnotatedTag(revision string) bool {
```

Comment:
Should we unit test this function individually? Seems like a fragile spot.

Reply:
It's covered by the end-to-end tests, but I agree that a unit test would be helpful.

Reply:
Added unit-tests, caught a bug :grin: